src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/data/ProfileRefreshQueueRepositoryIntegrationTest.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.reform.orgrolemapping.data;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.jdbc.Sql;
+import uk.gov.hmcts.reform.orgrolemapping.controller.BaseTestIntegration;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class ProfileRefreshQueueRepositoryIntegrationTest extends BaseTestIntegration {
+
+    @Autowired
+    private ProfileRefreshQueueRepository profileRefreshQueueRepository;
+
+    @Test
+    public void shouldInsertOrganisationProfileIds() {
+        List<String> organisationProfileIds = List.of("SOLICITOR_ORG", "DWP_GOV_ORG", "HMRC_GOV_ORG");
+        String orgProfileIds = String.join(",", organisationProfileIds);
+
+        profileRefreshQueueRepository.upsertOrganisationProfileIds(orgProfileIds, 1);
+
+        List<ProfileRefreshQueueEntity> profileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(profileRefreshQueueEntities, 3, 1);
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {
+        "classpath:sql/insert_organisation_profiles.sql"
+    })
+    public void shouldHandleDuplicateOrganisationProfileIds() {
+        List<ProfileRefreshQueueEntity> profileRefreshQueueEntities = profileRefreshQueueRepository.findAll();
+        validateData(profileRefreshQueueEntities, 1, 1);
+
+        List<String> organisationProfileIds = List.of("SOLICITOR_ORG");
+        String orgProfileIds = String.join(",", organisationProfileIds);
+
+        profileRefreshQueueRepository.upsertOrganisationProfileIds(orgProfileIds, 2);
```

Comment:
is it worth adding a test for when you dont supply the correct upsert version?

Reply:
the version that’s used in the upsert query will comes from the below query:

```
update access_types
set version = version + 1, access_types = ?::jsonb
returning version, access_types::text
```

so only when we update the above table, we then use the returned version value. but this table can only be updated once at a time due to the database lock in the below query (`for update`)

```
select version as version, access_types::text as
access_types
from access_types
for update
```

due to the database locking, the first update query should return a consistent and increasing upsert version